kani-compiler/src/codegen_cprover_gotoc/compiler_interface.rs

```diff
@@ -77,13 +78,16 @@ impl GotocCodegenBackend {
     }
 
     /// Generate code that is reachable from the given starting points.
+    ///
+    /// Invariant: iff `check_contract.is_some()` then `return.2.is_some()`
     fn codegen_items<'tcx>(
         &self,
         tcx: TyCtxt<'tcx>,
         starting_items: &[MonoItem],
         symtab_goto: &Path,
         machine_model: &MachineModel,
-    ) -> (GotocCtx<'tcx>, Vec<MonoItem>) {
+        check_contract: Option<InternalDefId>,
+    ) -> (GotocCtx<'tcx>, Vec<MonoItem>, Option<AssignsContract>) {
```

Comment:
I was wondering if there's any reason why you didn't add the AssignsContract to GotocCtx. Not a blocker.